gammapy/spectrum/energy_group_new.py

```diff
@@ -0,0 +1,37 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Spectrum energy bin grouping.
+"""
+import numpy as np
+import astropy.units as u
+
+class FluxPointBins:
+    def __init__(self, indices):
+        self.indices = indices
+
+    def lower_bounds(self):
+        return self.indices[:-1]
+
+    def upper_bounds(self):
+        return self.indices[1:] - 1
+
+    def bin(self, idx):
+        return np.arange(self.lower_bounds[idx],
+                         self.upper_bounds[idx])
+
+
+
+class FluxPointBinMaker:
+    def __init__(self, obs):
+        self.obs = obs
+        self.indices = None
+
+    def compute_bins_fixed(self, energy_binning):
```

Comment:
I think this class could be replaced by a single method.

Reply:
The rationale I was told was that in the future there will be other ways of defining bins without first going through the energy values. But the same thing can be accomplished just by overloading the init method, so I agree.